lib/OpenQA/WebAPI/Controller/Admin/Needle.pm

```diff
@@ -47,12 +47,43 @@ sub ajax {
     my ($self) = @_;
 
     my @conds;
+    my $params = {prefetch => qw/directory/};
 
+    # Only consider needles where the file is present
+    push(@conds, {file_present => 1});
+    my $total_needle_count = $self->db->resultset('Needles')->search({-and => \@conds}, $params);
+
+    # Conditions for search
+    my $search_value = $self->param('search[value]');
+    push(@conds, {filename => {-like => '%' . $search_value . '%'}}) if $search_value;
+
+    # Make columns directory.name, last_seen.t_created and last_match.t_created available by joining with JobModules
+    # Required for ordering by those columns and might also ease filtering
+    $params->{join} = [qw/directory last_seen last_match/];
+
+    # Parameter for order
+    my @columns = qw(directory.name filename last_seen.t_created last_match.t_created);
+    my @order_by_params;
+    my $index = 0;
+    while (1) {
+        my $column_index = $self->param("order[$index][column]");
+        my $column_order = $self->param("order[$index][dir]");
+        last
```

Comment:
please just forget that perl has post conditions. If you need 4 lines to break the if - a post condition is just not the right thing.